core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -161,6 +164,8 @@
 
     private final AtomicBoolean aborting = new AtomicBoolean(false);
 
+    private final Map<PlanNodeId, DynamicFilter> dynamicFilters;
```

Comment:
Does it mean there is just once chance to send DFs? If DFs were not present at `HttpRemoteTask` creation we would not send them later on?

Reply:
We're creating `dynamicFilters` in SqlStageExecution based on the DFs in the stage's plan fragment and that is passed down to HttpRemoteTask.
In HttpRemoteTask#sendUpdate we call DynamicFilter#getCurrentPredicate on the `dynamicFilters` entries to send the current predicate along with TaskUpdateRequest.